client/remotemodel.cpp

```diff
@@ -390,13 +390,23 @@ void RemoteModel::newMessage(const GammaRay::Message &msg)
                 qWarning() << "Unexpected empty index, probably some type failed to deserialize" << Q_FUNC_INFO;
                 continue;
             }
+            // in case of failure in reading itemData
+            // we will use this position to start looking
+            // for our special marker and restore the stream
+            // state
+            quint32 indexEndPos = msg.size();
```

Comment:
Seems wrong, this has to be "current pos", not "total size".

I think you need `Message::pos() { return payload().device()->pos(); }`  (and it could be used in message.cpp:269)

Reply:
Yep, done now. Pushed too early.